src/benchmarks/micro/runtime/Interop/ComWrappers.cs

```diff
@@ -0,0 +1,118 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Interop
+{
+    [BenchmarkCategory(Categories.Runtime)]
+    public unsafe class ComWrappersTests
+    {
+        static readonly ComWrappers s_instance = new DummyComWrappers();
+
+        static object QueryRCW(nint ptr)
+            => s_instance.GetOrCreateObjectForComInstance(ptr, CreateObjectFlags.None);
+
+        private List<nint> iterations;
+        private object targetRcw;
+
+        public ComWrappersTests()
+        {
+            // Allocate a simple IUnknown instance.
+            var ptr = (void**)NativeMemory.Alloc(5, (nuint)sizeof(void*));
+
+            ptr[1] = (void*)1; // Initial ref count.
+
+            // Define IUnknown vtable.
+            ptr[2] = (delegate* unmanaged<void*, Guid*, void**, int>)&QueryInterface;
```

Comment:
Or even better: Instead of creating your own bespoke COM implementation for the foreign COM object, you can use a second instance of ComWrappers to create the foreign COM object. I believe that it is going to test the same critical path - the two ComWrappers instances should not influence each other in any way.

Like:

```csharp
...
        static readonly ComWrappers s_foreignInstance = new DummyComWrappers();
...
            nint ptr = s_foreignInstance.GetOrCreateComInterfaceForObject(new object(), CreateComInterfaceFlags.None);

            // Transfer ownership to RCW instance. This populates the RCW cache.
            targetRcw = s_instance.GetOrCreateObjectForComInstance((nint)ptr, CreateObjectFlags.None);

            // Ownership was transferred.
            Marshal.Release((nint)ptr);
...
            protected override unsafe ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, out int count)
            {
                count = 0;
                return null;
            }
...
          // delete IID_IUnknown, QueryInterface, AddRef2, AddRef, Release
```